scripts/content_schema_artifacts.py

```diff
@@ -0,0 +1,143 @@
+HERE = Path(__file__).parent.resolve()
+SCHEMAS = HERE.parent / "src" / "content_schemas"
+SCHEMA_ARTIFACTS = HERE.parent / "src" / "content_schema_artifacts"
+
+
+def retrieve_from_filesystem(path: Path) -> Resource:
+    """
+    Retrieve a JSON object from a file.
+
+    Args:
+        path (Path): The path to the JSON file.
+
+    Returns:
+        Resource: A resource object representing the JSON content.
+
+    Raises:
+        FileNotFoundError: If the specified file does not exist.
+    """
+    if not path.exists():
+        raise FileNotFoundError(f"Schema file '{path}' not found")
+    with path.open("r", encoding="utf-8") as file:
+        contents = json.load(file)
+    return Resource(contents=contents, specification=DRAFT202012)
+
+
+def create_registry_from_filesystem(schema_dir: Path) -> Registry:
+    """
+    Create a registry of JSON objects obtained from the file system.
+
+    Args:
+        schema_dir (Path): The directory containing JSON schema files.
+
+    Returns:
+        Registry: A registry object containing resources parsed from JSON files.
+    """
+    my_registry = Registry()
+    for schema_file in schema_dir.glob("*.json"):
+        resource = retrieve_from_filesystem(schema_file)
+        my_registry = my_registry.__rmatmul__(resource)
+    return my_registry
+
+
+def modify_content(resource_content: dict, resolver: Resolver) -> tuple[dict, list]:
+    """
+    Modify the resource content by replacing referenced paths with corresponding JSON schemas.
+
+    Args:
+        resource_content (dict): The content of the resource to be modified.
+        resolver (Resolver): Resolver object used to look up referenced paths.
+
+    Returns:
+        dict: The modified resource content.
+    """
+    resolved_uris = []
+    if resource_content.get("properties"):
+        for key_, value_ in resource_content["properties"].items():
+            if isinstance(value_, dict) and "$ref" in value_.keys():
+                resource_content["properties"][key_] = resolver.lookup(
+                    value_["$ref"]
+                ).contents
+                resolved_uris.append(value_["$ref"])
+    return resource_content, resolved_uris
+
+
+def create_resource(
+    content: dict, specification: Specification = DRAFT202012
+) -> Resource:
+    """
+    Create a resource object from content and specification.
+    """
+    return Resource.from_contents(contents=content, default_specification=specification)
+
+
+def update_registry(my_registry: Registry, new_resource: Resource):
+    """
+    Update a resource in the registry.
+
+    Args:
+        my_registry (Registry): The registry to be updated.
+        new_resource (Resource): The new resource to be updated.
+
+    Returns:
+        Registry: The updated registry with the new resource.
+    """
+    modified_registry = my_registry.remove(URI(new_resource.id()))
+    return modified_registry.with_resource(
+        uri=URI(new_resource.id()), resource=new_resource
+    )
```

Comment:
Is this function necessary?

- Doesn't adding a resource with the same URI override the existing resource anyhow?
- Why are you using `with_resource` with manual URI specification, doesn't the schema have an ID?

Reply:
- True, it does overwrite. Fixed.  

- The function requires an URI as an input. It does not retrieve it from the resource id, because it was initially designed to add the resources without being crawled (https://github.com/python-jsonschema/referencing/blob/c0b176fd62e04018aff2872a42e720d8f2caf470/referencing/_core.py#L514). 

The function increases the readability of the code. I would keep it rather than merging its content to the main routine.